### PR TITLE
feat: support setting compression option

### PR DESCRIPTION
**Pull request by olavloite**
Adds support for setting the compressor to use for gRPC calls.

### CODE REVIEW

File: google-cloud-spanner/src/test/java/com/google/cloud/spanner/it/ITSpannerOptionsTest.java [hunk 1]
```diff
@@ -168,6 +168,27 @@ public void testMultipleSpannersFromSameSpannerOptions() throws InterruptedExcep
...
+    for (String compressorName : new String[] {"gzip", "identity", null}) {
+      SpannerOptions options =
+          env.getTestHelper().getOptions().toBuilder().setCompressorName(compressorName).build();
+      try (Spanner spanner = options.getService()) {
+        DatabaseClient client = spanner.getDatabaseClient(db.getId());
+        try (ResultSet rs =
+            client
+                .singleUse()
+                .executeQuery(Statement.of("SELECT 1 AS COL1 UNION ALL SELECT 2 AS COL2"))) {
+          assertThat(rs.next()).isTrue();
```
Comment (rven1) [inline google-cloud-spanner/src/test/java/com/google/cloud/spanner/it/ITSpannerOptionsTest.java:182]: Would it be possible to test an end to end compression and verify that the compressor is actually used.

Comment (skuruppu) [reply google-cloud-spanner/src/test/java/com/google/cloud/spanner/it/ITSpannerOptionsTest.java:182]: Yes absolutely. Once the support is in the staging environment, we can run this IT against it to test the change. @olavloite I'll give you a heads up when the feature is available in staging.

File: google-cloud-spanner/src/main/java/com/google/cloud/spanner/SpannerOptions.java [hunk 2]
```diff
@@ -558,6 +565,28 @@ public Builder setCallCredentialsProvider(CallCredentialsProvider callCredential
...
+    /**
+     * Sets the compression to use for all gRPC calls. The compressor must be a valid name known in
+     * the {@link CompressorRegistry}.
+     *
+     * <p>Supported values are:
+     *
+     * <ul>
+     *   <li>gzip: Enable gzip compression
+     *   <li>identity: Disable compression
+     *   <li><code>null</code>: Use default comporession
```
Comment (skuruppu) [inline google-cloud-spanner/src/main/java/com/google/cloud/spanner/SpannerOptions.java:577]:
```suggestion
     *   <li><code>null</code>: Use default compression
```

Comment (olavloite) [reply google-cloud-spanner/src/main/java/com/google/cloud/spanner/SpannerOptions.java]: Done.

**Comment by skuruppu** [comment]
@olavloite we can merge this change now. If you can resolve the conflict and merge, that would be great.

**Pull request merged**